### PR TITLE
[SYCL][Driver][thinLTO] Error for -foffload-lto and -fsycl-device-code-split=off

### PR DESCRIPTION
**Pull request by sarnex**
This is really complicated to support so we decided to disallow it for now.

I also fixed the condition to be more of what we want.

### CODE REVIEW

**Comment by sarnex** [comment]
Gen12 failure unrelated, https://github.com/intel/llvm/issues/14186

File: clang/test/Driver/sycl-lto.cpp [hunk 1]
```diff
@@ -4,6 +4,10 @@
 // RUN: not %clangxx -fsycl -foffload-lto=thin %s -### 2>&1 | FileCheck -check-prefix=CHECK_ERROR %s
 // CHECK_ERROR: unsupported option '-foffload-lto=thin' for target 'spir64-unknown-unknown'
 
+// Verify we error when using the new offload driver but with device code split set to off.
+// RUN: not %clangxx -fsycl --offload-new-driver -foffload-lto=thin -fsycl-device-code-split=off %s -### 2>&1 | FileCheck -check-prefix=CHECK_SPLIT_ERROR %s
+// CHECK_SPLIT_ERROR: unsupported option '-foffload-lto=thin' for target 'spir64-unknown-unknown'
```
Comment (asudarsa) [inline clang/test/Driver/sycl-lto.cpp:9]: Seems like the error message will need to specify that the error is due to combined use of -foffload-lto=thin and -fsycl-device-code-split=off?

Comment (sarnex) [reply clang/test/Driver/sycl-lto.cpp:9]: Sure, I will add a better error, thanks.

Comment (sarnex) [reply clang/test/Driver/sycl-lto.cpp]: fixed in latest commit, thx

**Review by mdtoguchi — APPROVED** [review]

**Review by asudarsa — APPROVED** [review]
LGTM. Thanks. 
Please add [ThinLTO] to PR header.
Also, the description says, 'disallow for now'. Do we expect to add this support at a later stage? Do we need a TODO to track this? May be an internal trcker?

Thanks

**Review by smanna12 — APPROVED** [review]
FE change LGTM

**Comment by sarnex** [comment]
@asudarsa Thanks, will update title. I do not plan to add support unless we get an important customer use case, so I don't think we need a TODO

**Pull request merged**